app/code/community/Vindi/Subscription/Trait/PaymentMethod.php

```diff
@@ -245,7 +245,8 @@ protected function processSinglePayment($payment, $order, $customerId)
             if ($body['payment_method_code'] === "bank_slip"
                 || $body['payment_method_code'] === "debit_card"
                 || $bill['status'] === "paid"
-                || $bill['status'] === "review") {
+                || $bill['status'] === "review")
+                || $bill['charges'][0]['status'] === "fraud_review") {
```

Comment:
Aqui é sempre a primeira cobrança? Isso é garantido?

Reply:
Nesse caso sim, pois a cobrança é realizada imediatamente após a compra.
O que pode existir em alguns casos são várias transações, porém são elas dentro (do status) da mesma cobrança :smile:

Reply:
Mas a cobrança não pode ser cancelada por algum outro motivo e ser criada uma outra?

Reply:
Nesse momento não, esse é o tratamento do retorno da criação de uma _Bill_ (fatura).

Reply:
Entendi. Nesse caso eu só tentaria deixar mais bunitinho heheh no php não tem um método pra pegar o primeiro ou o último elemento de um array? Tipo: `end($bill['charges'])['status']` ou `reset($bill['charges']['status'])`.

Reply:
Eu preciso da primeira posição do _array_, o que seria muito legal se houvesse um _first()_ ou _begin()_ :fearful: 
O reset nesse caso retorna a primeira posição da **Bill**, porém eu preciso da primeira posição da **Charge** que está dentro da **Bill**, um jeito de usá-lo seria criar uma variável para receber o _array_ Charges.

Particularmente o jeito mais bonito que achei foi esse; Mas se preferir posso usar esse:
```` array_shift($bill['charges'])['status'] ````

Reply:
Foi só uma sugestão mesmo, não manjo muito de PHP. Mas tá aprovado do jeito que está. o/

Reply:
Sem problemas @dmente03 :smile: 
O ruim do [array_shift](http://php.net/manual/pt_BR/function.array-shift.php) é que ele remove a primeira posição do array e se eu for precisar utilizar o _Bill_ em outro momento, estará incompleto (sem _Charges_).

Vou manter então a primeira opção :+1:

Reply:
Não tem maneira bonita de fazer isso, daria para por exemplo `current($bill['charges'])['status']`, e mesmo assim não é nada legível. Coisas do PHP :) (PS: ao invés de usar `array_shift` use `current` que não é destrutivo)